### PR TITLE
feat(@clayui/css): Buttons adds a 24px tall button, `btn-xs`

### LINKED ISSUE #5081: New 24px Button + Rounded property
## What is your proposal?
Include a new 24px button size. To avoid breaking changes, this variant will be named XS Button since there is already a Regular (40px) and Small (32px), in the next major version it would be nice to have this size for the buttons renamed to: Small, Regular, and Large. 
We will also allow having a property for Rounded buttons. Ideally, we won't be providing a free border-radius but only allowing 2 styles:
- Smooth (Default of 4px, what we currently have)
- Rounded (fully rounded buttons) 
image

**The rounded property should also be allowed for bigger buttons for consistency** 

## Why would adopting this proposal be beneficial?
We have a product team request from Echo and we've also analyzed the usage in portal finding another case from the Tango team, and more in general in Portal. Also, this is something that Dialect found missing in Lexicon/Clay buttons. 

## What are the alternatives to this proposal?
none

## More info
[Request from Echo ](https://issues.liferay.com/browse/DESIGN-4172)
[Documentation](https://liferay.atlassian.net/wiki/spaces/ENGLEXICON/pages/2121269264/DESIGN-4172+New+24px+Button+Variant) (I'm still working on this but the specs and technical aspects are covered, I need to fill out the usage recommendations for designers).
[Figma file](https://www.figma.com/file/nTRgW4NlyqPJc2rtdmLNdg/design-4172-new-button-variant?node-id=65%3A3538)

### PR DESCRIPTION
**Pull request by pat270**
fixes #5081

I added the `size` and `shape` attributes to `ClayButton`, hopefully these make sense.

### CODE REVIEW

**Review by matuzalemsteles — COMMENTED** [review]
Hey @pat270 I just added some comments, it would also be nice to add in the `BREAKING.md` file of the package the information that we will be changing the default size and readjusting the sizes for the next major version. This can look like a list in the file, see example in other packages.

File: packages/clay-button/src/Button.tsx [hunk 1]
```diff
@@ -50,6 +58,16 @@ interface IProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
 	 */
 	outline?: boolean;
 
+	/**
+	 * Determines the border radius of a link.
+	 */
+	shape?: 'rounded';
```
[the remark this answers is not in the document]
Comment (pat270) [reply packages/clay-button/src/Button.tsx:64]: @matuzalemsteles what if we called it `radius`? I'd like to leave it as open ended as possible. I can foresee different types of rounded or square buttons in the future.

[the remark this answers is not in the document]
Comment (matuzalemsteles) [reply packages/clay-button/src/Button.tsx:64]: @pat270 probably `radius` might fit better here, I still prefer to keep only the `rounded` API and if in the future we will have more radius types we can add the API to handle all types of radius, as was done for `size` now. But feel free to decide which direction to go, I'm fine with both.

[the remark this answers is not in the document]
Comment (pat270) [reply packages/clay-button/src/Button.tsx:64]: Np, we can use rounded and change in the future.

**Comment by pat270** [comment]
@matuzalemsteles this should be gtg if the tests pass.

**Review by matuzalemsteles — APPROVED** [review]
LGTM! Thanks @pat270!

**Pull request merged**